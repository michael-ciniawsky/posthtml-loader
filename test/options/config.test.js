'use strict'

const webpack = require('../helpers/compiler')

describe('Options', () => {
  describe('config', () => {
    test('path - {String}', () => {
      const config = {
        loader: {
          test: /\.html$/,
          options: {
            config: {
              path: 'test/fixtures/posthtml.config.js'
            }
          }
        }
      }

      return webpack('fixture.js', config)
        .then((stats) => {
          const [module] = stats.toJson().modules

          expect(module.source).toMatchSnapshot()
        })
        .catch((err) => err)
    })

    test('ctx - {Object}', () => {
      const config = {
        loader: {
          test: /\.html$/,
          options: {
            config: {
              path: 'test/fixtures/options/config/posthtml.config.js',
              ctx: { plugin: true }
            }
          }
        }
      }

      return webpack('fixture.js', config)
        .then((stats) => {
          const [module] = stats.toJson().modules

          expect(module.source).toMatchSnapshot()
        })
        .catch((err) => err)
    })
  })
})
