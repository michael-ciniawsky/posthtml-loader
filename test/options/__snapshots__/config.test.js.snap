// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options config ctx - {Object} 1`] = `
"export default \`<section>Hello</section>
\`"
`;

exports[`Options config path - {String} 1`] = `
"export default \`<section>Hello</section>
\`"
`;
